tests/AddCommandTest.php

```diff
@@ -192,4 +192,36 @@ public function it_adds_win_bash_compat_if_the_force_windows_option_is_passed()
             $this->assertEquals(strpos($content, "#!/bin/bash"), 0);
         }
     }
+
+    /**
+     * @test
+     */
+    public function it_handles_commands_defined_in_an_array()
+    {
+        $gitDir = 'test-git-dir';
+        $hookDir = "{$gitDir}/hooks";
+
+        $hooks = [
+            'test-pre-commit' => [
+                'echo pre-commit first',
+                'echo pre-commit second',
+                'echo pre-commit third',
+            ],
+        ];
+
+        $command = new AddCommand($hooks);
+        $commandTester = new CommandTester($command);
+
+        $commandTester->execute(['--git-dir' => $gitDir]);
+
+        foreach ($hooks as $hook => $scripts) {
+            $content = file_get_contents($hookDir . '/' . $hook);
+
+            // Keep the last lines of the hook file (as many lines as script command)
+            $lines = explode(PHP_EOL, $content);
+            $lines = array_slice($lines, - count($scripts));
```

Comment:
We don't need to keep as many lines as the script command because it should only ever have those many lines. If for some reason they don't match, then the test should fail. You can just remove this and the explode part

Reply:
I was thinking about making a little assumptions as possible here.

> [...] because it should only ever have those many lines [...]

Unless i misunderstood something, it's not the case on Windows (commands are prepended with the shebang).

And in the future, if it's decided to add a blank line between the shebang and the commands, we won't have to rewrite the test (actually, adding that blank line shouldn't cause the test to fail).

Reply:
You're right about the shebang. However, this current way is still a little brittle. If we decided to add a newline at the end of the every file, this test will fail. Let's change the test condition to 
```php
$this->assertContains(implode(PHP_EOL, $scripts), $content);
```

That way, it should always work

Reply:
>  If we decided to add a newline at the end of the every file, this test will fail

That's right. I'll fix the test later :)